asv/results.py

```diff
@@ -559,19 +559,41 @@ def load(cls, path, machine_name=None):
 
         return obj
 
+    def _merge_results(self, old):
+        # Merge existing results with old results
+        # When benchmark is parameterized and new result is either failed or
+        # skipped for a given param combination then keep old result.
+        new_results = self._results
+        old_results = old._results
+        for key, val in six.iteritems(old_results):
+            if key not in new_results:
+                new_results[key] = val
+            elif self._benchmark_params[key]:
+                old_benchmark_results = {}
+                for idx, param_set in enumerate(itertools.product(
+                        *old._benchmark_params[key])):
+                    old_benchmark_results[param_set] = val[idx]
+                for idx, param_set in enumerate(itertools.product(
+                        *self._benchmark_params[key])):
+                    if (new_results[key][idx] is None and
+                            old_benchmark_results.get(param_set) is not None):
```

Comment:
I think this logic should use the mask stored in `self._benchmark_selection`, not look at the result values.

Benchmarks may be skipped also due to the benchmark suite itself, which is a somewhat different situation than here.

Reply:
Ok, sorry, that information is not available here.

Then I guess the next best is to replace this if statement with `if util.is_nan(new_results[key][idx]):`, which will fill in skipped results but leave failures in.